src/seedu/addressbook/data/person/Address.java

```diff
@@ -37,20 +56,42 @@ public static boolean isValidAddress(String test) {
     }
 
     @Override
-    public String toString() {
-        return value;
+    public String toString() {    	
+    	
+    	String temp = "";
+    	
+    	if (block != null) {
```

Comment:
If we ensure that illegal address (e.g. with missing parts) cannot be constructed in the first place, then all the null pointers check can be avoided.